gateway-server/src/main/java/org/apache/knox/gateway/services/topology/impl/DefaultTopologyService.java

```diff
@@ -901,8 +906,8 @@ public void onFileChange(File file) {
     public boolean accept(File file) {
       boolean accept = false;
       if (!file.isDirectory() && file.canRead()) {
-        String extension = FilenameUtils.getExtension(file.getName());
-        if (SUPPORTED_EXTENSIONS.contains(extension)) {
+        String fileName = file.getName();
+        if (isDescriptorFile(fileName) || isXmlTypeDescriptorFile(fileName)) {
```

Comment:
I thought there was going to be a separate monitor for this CM integration descriptor, the handling of which would be contained in the cm-integration module.

Reply:
I think we agreed to use the existing monitor. On the other hand, it's better separated if monitoring was in the new cm-integration module. Let me do it there.

Reply:
Done